src/shell/dotnet/Shell/Shell.csproj

```diff
@@ -47,16 +47,31 @@
 	</ItemGroup>
 
 	<PropertyGroup>
-		<Fdc3BundleOutput>$(ComposeUIRepositoryRoot)/src/shell/js/composeui-fdc3/dist/fdc3-iife-bundle.js</Fdc3BundleOutput>
+		<Fdc3BundleOutput>$(ComposeUIRepositoryRoot)\src\shell\js\composeui-fdc3\dist\fdc3-iife-bundle.js</Fdc3BundleOutput>
 	</PropertyGroup>
 
 	<ItemGroup>
-		<EmbeddedResource Include="$(Fdc3BundleOutput)">
-		</EmbeddedResource>
+		<EmbeddedResource Include="$(Fdc3BundleOutput)" />
 	</ItemGroup>
 
-	<Target Name="PreBuild" BeforeTargets="PrebuildEvent" Condition="!Exists('$(Fdc3BundleOutput)')">
-		<Exec Command="cd $(ComposeUIRepositoryRoot)" />
-		<Exec Command="npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
+	<Target Name="InstallMonoRepoDependencies"
+		Condition="!Exists('$(ComposeUIRepositoryRoot)\node_modules')">
+		<Message Importance="high" Text="Installing node_modules..." />
+		<Exec Command="cd $(ComposeUIRepositoryRoot) &amp; npm install" />
+		<Message Importance="normal" Text="Finished command: npm install." />
 	</Target>
+
+	<Target Name="RunFdc3LernaBuild">
+		<Message Importance="high" Text="Running npx lerna run build for composeui-fdc3..." />
+		<Exec Command="cd $(ComposeUIRepositoryRoot) &amp; npx lerna run build --scope @morgan-stanley/composeui-fdc3" />
+		<Message Importance="normal" Text="Finished building the fdc3 bundle." />
+	</Target>
+
+	<Target
+		Name="BuildComposeUIFdc3Bundle"
+		BeforeTargets="PrebuildEvent">
+		<CallTarget Targets="InstallMonoRepoDependencies" />
```

Comment:
This won't be called if the node_modules folder exists, but the other target will be executed.
The Choose and When and OtherWise cannot be used within a Target.